static/app/views/dashboards/widgetCard/autoSizedText.tsx

```diff
@@ -0,0 +1,177 @@
+    const childDimensions = getElementDimensions(childRef.current);
+
+    if (!parentDimensions || !childDimensions) {
+      // Refs are not ready or cannot be measured, abandon
+      return;
+    }
+
+    // Calculate the width and height disparity between the child and parent. A disparity of 0 means they're the same size.
+    const widthDisparity = calculateDimensionDisparity(
+      childDimensions,
+      parentDimensions,
+      'width'
+    );
+
+    const heightDisparity = calculateDimensionDisparity(
+      childDimensions,
+      parentDimensions,
+      'height'
+    );
+
+    const childFitsIntoParent =
+      childDimensions.width <= parentDimensions.width &&
+      childDimensions.height <= parentDimensions.height;
+
+    if (
+      childFitsIntoParent &&
+      (widthDisparity <= MAXIMUM_DISPARITY || heightDisparity <= MAXIMUM_DISPARITY)
+    ) {
+      // The child fits completely into the parent _and_ at least one dimension is very similar to the parent size (i.e., it fits nicely in the parent). Abandon, we're done!
+      return;
+    }
+
+    if (
+      childDimensions.width > parentDimensions.width ||
+      childDimensions.height > parentDimensions.height
+    ) {
+      // The element is bigger than the parent, scale down
+      const newFontSize = (fontSizeLowerBound + fontSize) / 2;
+
+      startTransition(() => {
+        setCalculationCount(previousCalculationCount => previousCalculationCount + 1);
+        setFontSizeUpperBound(fontSize);
+        setFontSize(newFontSize);
+      });
+    } else if (
+      childDimensions.width < parentDimensions.width ||
+      childDimensions.height < parentDimensions.height
+    ) {
+      // The element is too small, scale up
+      const midpoint = (fontSizeUpperBound + fontSize) / 2;
+
+      startTransition(() => {
+        setCalculationCount(previousCalculationCount => previousCalculationCount + 1);
+        setFontSizeLowerBound(fontSize);
+        setFontSize(midpoint);
+      });
+    }
+  }, [
+    fontSize,
+    minFontSize,
+    fontSizeLowerBound,
+    maxFontSize,
+    fontSizeUpperBound,
+    calculationCount,
+    calculationCountLimit,
+    parentHeight,
+    parentWidth,
+  ]);
+
+  return (
+    <ParentSizeMimic ref={parentRef}>
+      <SizedChild ref={childRef} style={{fontSize}}>
+        {children}
+      </SizedChild>
+    </ParentSizeMimic>
+  );
+}
+
+const ParentSizeMimic = styled('div')`
+  height: 100%;
+  width: 100%;
+  overflow: hidden;
+  position: relative;
+`;
+
+const SizedChild = styled('div')`
+  display: inline-block;
+`;
+
+const RESIZE_DEBOUNCE = 200; // ms
+const DEFAULT_CALCULATION_COUNT_LIMIT = 5;
+const MAXIMUM_DISPARITY = 0.05;
+
+type Dimensions = {
+  height: number;
+  width: number;
+};
+
+function calculateDimensionDisparity(
+  a: Dimensions,
+  b: Dimensions,
+  dimension: 'height' | 'width'
+): number {
+  return 1 - Math.abs(a[dimension] / b[dimension]);
```

Comment:
```suggestion
  return Math.abs(1 - (a[dimension] / b[dimension]));
```

Should the `Math.abs` be wrapped around the subtraction? Otherwise this looks like it can produce disparities < 0. Seems like resizing still works anyways because we do a check to make sure child dimensions are less than or equal to the parent.